affinity/cpp-20/d0796r2.md

```diff
@@ -170,150 +187,134 @@ This feature could be easily scaled to heterogeneous and distributed systems, as
-
 ### Binding execution and allocation to resources
 
-When creating an `execution_context` from a given `execution_resource`, the executors and allocators associated with it are bound to that `execution_resource`. For example: when creating an `execution_resource` from a CPU socket resource, all executors associated with the given socket will spawn execution agents with affinity to the socket partition of the system *(Listing 5)*.
-
-```cpp
-auto cList = std::execution::this_system::resources();
-// FindASocketResource is a user-defined function that finds a 
-// resource that is a CPU socket in the given resource list
-auto& socket = findASocketResource(cList);
-execution_contexteC{socket} // Associated with the socket
-auto executor = eC.executor(); // By transitivity, associated with the socket too
-auto socketAllocator = eC.allocator(); // Retrieve an allocator to the closest memory node
-std::vector<int, decltype(socketAllocator)> v1(100, socketAllocator);
-std::generate(par.on(executor), std::begin(v1), std::end(v1), std::rand);
-```
-*Listing 5: Example of allocating with affinity to an execution resource*
-
-The construction of an `execution_context` on a component implies affinity (where possible) to the given resource. This guarantees that all executors created from that `execution_context` can access the resources and the internal data structures requires to guarantee the placement of the processor.
-
-Only developers that care about resource placement need to care about obtaining executors and allocations from the correct `execution_context` object. Existing code for vectors and STL (including the Parallel STL interface) remains unaffected.
+An execution context, such as a thread pool, may be bound to an execution resource.
+Binding enables multiple execution contexts to be created and bound to disjoint
+execution resources so that their respective execution agents do not compete for
+their respective execution resources.
 
-If a particular policy or algorithm requires to access placement information, the resources associated with the passed executor can be retrieved via the link to the `execution_context`.
 
 ## Header `<execution>` synopsis
 
     namespace std {
     namespace experimental {
     namespace execution {
 
-    /* Execution resource */
+    /* Execution resource capable of executing std::thread */
 
-    class execution_resource {
+    class thread_execution_resource {
      public:
+      using iterator = /* implementation defined, dereferences to thread_execution_resource */ ;
 
-      execution_resource() = delete;
-      execution_resource(const execution_resource &);
-      execution_resource(execution_resource &&);
-      execution_resource &operator=(const execution_resource &);
-      execution_resource &operator=(execution_resource &&);
-      ~execution_resource();
+      ~thread_execution_resource();
+      thread_execution_resource() = delete;
```

Comment:
I think we decided to make the execution resource type copyable and moveable so that it would be easier to use them in standard algorithms.

Reply:
I thought this was for the `std::vector<execution_context>` interface, and with iterate-able `thread_execution_resource` was not longer necessary.

Reply:
That's true, but we decided that we should keep resources copyable and moveable so that they could be used in standard algorithms, in order to for example sort by affinity or find particular resources. Although this is something we will need to investigate further. Something like this:

```cpp
std::sort(execRes.begin(), execRes.end(), [&memRes](const foo &lhs, const foo &rhs){
  return affinity_query<read, latency>(lhs, memRes) > affinity_query<read, latency>(rhs, memRes);
});
```